main.py

```
@@ -1818,10 +1827,19 @@ async def telegram_webhook_handler(request):
-                    logger.error(f"Failed to process update: {exc}")
-                    return web.Response(status=500, text="Failed to process update")
+                    logger.error(f"Failed to parse update: {exc}")
+                    return web.Response(status=400, text="Invalid update")
+
+                # Hand the update to PTB's own processing queue and acknowledge immediately.
+                # Awaiting process_update() here would keep the HTTP request open for the
+                # whole handler; anything slower than Telegram's webhook timeout makes
+                # Telegram redeliver the same update, which shows up as duplicate replies.
+                try:
+                    await self.application.update_queue.put(update)
+                except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)